src/test/java/com/google/api/codegen/testdata/csharp_gapic_resourcenames_library.baseline

```diff
@@ -0,0 +1,719 @@
+============== file: Google/Example/Library/V1/ResourceNames.cs ==============
+// Copyright 2016, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Generated code. DO NOT EDIT!
+
+using Google.Api.Gax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Example.Library.V1
+{
+    public interface IResourceName{}// TODO: This will be in GAX
+    public interface ISingleResourceName:IResourceName{} // TODO: Is this required?
+
+    public sealed class UnknownResourceName : IResourceName // TODO: This will be in GAX
+    {
+        public static UnknownResourceName Parse(string value) => new UnknownResourceName(value);
+
+        public static bool TryParse(string value, out UnknownResourceName result)
+        {
+            result = new UnknownResourceName(value);
+            return true;
+        }
+
+        public UnknownResourceName(string value)
+        {
+            Value = GaxPreconditions.CheckNotNull(value, nameof(value));
+        }
+
+        public string Value { get; }
+
+        public override string ToString() => Value;
+    }
+
+    public sealed class ResourceNameList<TResource> : IList<TResource> where TResource : IResourceName // TODO: This will be in GAX
```

Comment:
This is in GAX so not as relevant, but let's call it TName rather than TResource. I'd like to maintain a clear separation between resources and resource names wherever possible.